src/main/webapp/3Dscript.js

```diff
@@ -45,144 +49,285 @@ function initThreeJs() {
-    .then(response => response.json())
-    .then(data => pose = data)
-    .then(() => addMap());
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const id = urlParams.get('id');
+  const type = urlParams.get('dataType');
+  fetch('/getrun?id=' + id + '&dataType=' + type)
+      .then((response) => response.json())
+      .then((data) => pose = data)
+      .then(() => addMap());
+}
+
+/**
+ * Finds the median value of the latitude, longitude,
+ * and altitude. The median is used as a reference point
+ * for the East-North-Up (ENU) conversion and as the center
+ * point for the 2D map.
+ * @return {array} An array containing median lat, lng, and alt
+ *   coordinates.
+ */
+function findMedians() {
+  const arrays = {lat: [], lng: [], alt: []};
+  for (const point of pose) {
+    arrays.lat.push(point.lat);
+    arrays.lng.push(point.lng);
+    arrays.alt.push(point.alt);
+  }
+  const sorted = {
+    lat: [...arrays.lat].sort((a, b) => a - b),
+    lng: [...arrays.lng].sort((a, b) => a - b),
+    alt: [...arrays.alt].sort((a, b) => a - b)};
+  const mid = {
+    lat: sortedLat[Math.floor(sorted.lat.length / 2)],
+    lng: sortedLng[Math.floor(sorted.lng.length / 2)],
+    alt: sortedAlt[Math.floor(sorted.alt.length / 2)]};
+  return [mid.lat, mid.lng, mid.alt];
 }
 
 /**
  * Creates the 2D terrain and points a light at it.
- * It also calls two other functions so it can wait on the fetch 
- * request and significantly cut down rendering time with 
+ * It also calls two other functions so it can wait on the fetch
+ * request and significantly cut down rendering time with
  * geometry instancing.
 */
 function addMap() {
+  const [medianLat, medianLng, medianAlt] = findMedians();
+  median = {lat: medianLat, lng: medianLng, alt: medianAlt};
+
   // Creates the 2D map.
-  let loader = new THREE.TextureLoader();
-  let material = new THREE.MeshLambertMaterial({ 
+  const loader = new THREE.TextureLoader();
+  const material = new THREE.MeshLambertMaterial({
     map: loader.load(
-      'https://maps.googleapis.com/maps/api/staticmap' +
-      '?format=png&center=' + pose[0].lat + ',' + pose[0].lng + 
-      '&zoom=18&size=500x500&key=' + apiKey)
-    });
-  let geometry = new THREE.PlaneGeometry(50,50);
+        'https://maps.googleapis.com/maps/api/staticmap' +
+      '?format=png&center=' + median.lat + ',' + median.lng +
+      '&zoom=18&size=500x500&key=' + apiKey),
+  });
+  const geometry = new THREE.PlaneGeometry(200, 200);
   const map = new THREE.Mesh(geometry, material);
-  map.position.set(0,-4,0);
-  map.rotation.x = -Math.PI / 2;
+  map.position.set(0, 0, -4);
   scene.add(map);
 
   // Add the light to the scene.
-  let light = new THREE.PointLight(0xffffff, 1, 0 );
-  light.position.set(0, 100, 0);
-  scene.add(light)
+  const light = new THREE.PointLight(0xffffff, 1, 0 );
+  light.position.set(0, 0, 1000);
+  scene.add(light);
+
+  createInstances();
```

Comment:
it doesnt really make sense to me why add map would need to call these functions.  could there be a funciton in fetch data that calls them? seems a bit more logic friendly, tho i see it doesnt change the code fundamentally